### DISCUSSION TITLE
TypeError: invalid file: PosixPath('/usr/local/searx/searx/data/engines_languages.json')

### DISCUSSION THREAD

**Discussion by mistersixt** [searx/searx#2464] [Installation issue: custom]
Hi there, 

I just did an "git pull origin master" (the previous pull is maybe 4 or 5 days ago), and now I got this upon the initial start (Debian 9 this is, with python 3.5):

Thu Jan 14 10:17:27 2021 - *** Starting uWSGI 2.0.14-debian (64bit) on [Thu Jan 14 10:17:27 2021] ***
Thu Jan 14 10:17:27 2021 - compiled with version: 6.3.0 20170516 on 30 August 2020 17:14:04
Thu Jan 14 10:17:27 2021 - os: Linux-4.9.0-14-amd64 #1 SMP Debian 4.9.246-2 (2020-12-17)
Thu Jan 14 10:17:27 2021 - nodename: mail
Thu Jan 14 10:17:27 2021 - machine: x86_64
Thu Jan 14 10:17:27 2021 - clock source: unix
Thu Jan 14 10:17:27 2021 - pcre jit disabled
Thu Jan 14 10:17:27 2021 - detected number of CPU cores: 2
Thu Jan 14 10:17:27 2021 - current working directory: /
Thu Jan 14 10:17:27 2021 - writing pidfile to /run/uwsgi/app/searx/pid
Thu Jan 14 10:17:27 2021 - detected binary path: /usr/bin/uwsgi-core
Thu Jan 14 10:17:27 2021 - setgid() to 1028
Thu Jan 14 10:17:27 2021 - setuid() to 1028
Thu Jan 14 10:17:27 2021 - chdir() to /usr/local/searx/searx/
Thu Jan 14 10:17:27 2021 - your processes number limit is 31863
Thu Jan 14 10:17:27 2021 - your memory page size is 4096 bytes
Thu Jan 14 10:17:27 2021 - detected max file descriptor number: 1024
Thu Jan 14 10:17:27 2021 - lock engine: pthread robust mutexes
Thu Jan 14 10:17:27 2021 - thunder lock: disabled (you can enable it with --thunder-lock)
Thu Jan 14 10:17:27 2021 - uwsgi socket 0 bound to UNIX address /run/uwsgi/app/searx/socket fd 3
Thu Jan 14 10:17:27 2021 - Python version: 3.5.3 (default, Nov 18 2020, 21:09:16)  [GCC 6.3.0 20170516]
Thu Jan 14 10:17:27 2021 - Set PythonHome to /usr/local/searx/searx-ve/
Thu Jan 14 10:17:27 2021 - Python main interpreter initialized at 0x556b135ac1a0
Thu Jan 14 10:17:27 2021 - python threads support enabled
Thu Jan 14 10:17:27 2021 - your server socket listen backlog is limited to 100 connections
Thu Jan 14 10:17:27 2021 - your mercy for graceful operations on workers is 60 seconds
Thu Jan 14 10:17:27 2021 - mapped 363840 bytes (355 KB) for 4 cores
Thu Jan 14 10:17:27 2021 - *** Operational MODE: preforking ***
Thu Jan 14 10:17:27 2021 - added /usr/local/searx/ to pythonpath.
Thu Jan 14 10:17:27 2021 - spawned uWSGI master process (pid: 30854)
Thu Jan 14 10:17:27 2021 - spawned uWSGI worker 1 (pid: 30860, cores: 1)
Thu Jan 14 10:17:27 2021 - spawned uWSGI worker 2 (pid: 30861, cores: 1)
Thu Jan 14 10:17:27 2021 - spawned uWSGI worker 3 (pid: 30862, cores: 1)
Thu Jan 14 10:17:27 2021 - spawned uWSGI worker 4 (pid: 30863, cores: 1)
Traceback (most recent call last):
  File "/usr/local/searx/searx/webapp.py", line 62, in <module>
    from searx.engines import (
  File "/usr/local/searx/searx/engines/__init__.py", line 28, in <module>
Traceback (most recent call last):
  File "/usr/local/searx/searx/webapp.py", line 62, in <module>
    from searx.engines import (
  File "/usr/local/searx/searx/engines/__init__.py", line 28, in <module>
    from searx.data import ENGINES_LANGUAGES
  File "/usr/local/searx/searx/data/__init__.py", line 24, in <module>
    ENGINES_LANGUAGES = load('engines_languages.json')
  File "/usr/local/searx/searx/data/__init__.py", line 11, in load
    with open(data_dir / filename, encoding='utf-8') as fd:
TypeError: invalid file: PosixPath('/usr/local/searx/searx/data/engines_languages.json')
Thu Jan 14 10:17:29 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Thu Jan 14 10:17:29 2021 - *** no app loaded. going in full dynamic mode ***
    from searx.data import ENGINES_LANGUAGES
  File "/usr/local/searx/searx/data/__init__.py", line 24, in <module>
    ENGINES_LANGUAGES = load('engines_languages.json')
...

...ending up with "internal server error" in the webbrowser. Any help is welcome (for now I restored a backup).

Kind regards, mistersixt.

**Comment by dalf** [comment]
Hi,

Python 3.5 has reached end of life since September 2020: https://www.python.org/downloads/release/python-3510/

Searx dropped the Python 3.5 support since this commit: https://github.com/searx/searx/commit/f5c3cb7afafc3d25380e681230263320ddcbbdfa : different incoming features will require Python 3.6.

I don't see a backport from Debian Buster, is [pyenv](https://github.com/pyenv/pyenv) an option for you ?
